xarray/plot/utils.py

```diff
@@ -216,8 +216,13 @@ def _determine_cmap_params(
         vlim = abs(vmax - center)
 
     if possibly_divergent:
+        levels_are_divergent = (
+            isinstance(levels, Iterable) and levels[0] * levels[-1] < 0
```

Comment:
I don't know this well but do we know levels are monotonic or do we need to sort first?

Reply:
nice catch. matplotlib raises an error if `levels` is not sorted using this check: `np.min(np.diff(self.levels)) <= 0.0`. Right now, I'm doing `levels = sorted(levels)`. Should we raise an error instead?

Reply:
No strong view, either is reasonable imo...